### PR TITLE
Feature/resolving technical debt

### PR DESCRIPTION
**Pull request by Guillaume1868**

### CODE REVIEW

File: src/models/router.php
Comment (Joumoune) [inline src/models/router.php:1]: use error function instead of lost

File: src/models/field.php [hunk 1]
```diff
@@ -21,7 +21,7 @@ public function __construct(int $id)
 		$this->database_access = (new DatabasesChoose())->getDatabase();
 
 		if (!$this->database_access->doesFieldExist($id)) {
-			throw new Exception('Field Does Not Exist');
+			throw new FieldNotFoundException('Field Does Not Exist');
```
Comment (Joumoune) [inline src/models/field.php]: no need to specify error message because it is already set like this by default

**Review by Joumoune — CHANGES_REQUESTED** [review]

**Review by Joumoune — APPROVED** [review]

File: src/models/exercise.php [hunk 2]
```diff
@@ -99,3 +99,12 @@ public function getFieldsCount(): int
 		return $this->database_access->getFieldsCount($this->id);
 	}
 }
+
+class ExerciseNotFoundException extends LooperException
+{
+	public function __construct($message = 'The exercise does not exist', $code = 0, Exception $previous = null)
+	{
+		// Make sure everything is assigned properly
```
Comment (EthannSchneider) [inline src/models/exercise.php:107]: not usefull comment pls delete it

File: src/models/field.php (hunk 1 above)
Comment (EthannSchneider) [inline src/models/field.php]: remove args since it's the same as the default one

File: src/models/exceptions/looper_exception.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+class LooperException extends Exception
+{
+	private int $httpReturnCode = 0;
+	private string $httpErrorMessage = '';
+
+	public function __construct(int $httpReturnCode, string $httpErrorMessage, $message = '', $code = 0, Exception $previous = null)
+	{
+		// Make sure everything is assigned properly
```
Comment (EthannSchneider) [inline src/models/exceptions/looper_exception.php]: remove unused comment

File: src/views/errors/error.php [hunk 4]
```diff
@@ -1,12 +1,12 @@
 <?php
 $title = 'Bad Request';
```
Comment (EthannSchneider) [inline src/views/errors/error.php]: pls use a var to assign this value

**Review by EthannSchneider — CHANGES_REQUESTED** [review]

File: src/models/exceptions/looper_exception.php (hunk 3 above)
Comment (EthannSchneider) [reply src/models/exceptions/looper_exception.php]: maybe there is other unused comment pls remove them

**Review by EthannSchneider — APPROVED** [review]

**Pull request merged**